libraries/file_helpers.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+module CernerSplunk
+  # Helper methods for file manipulation
+  module FileHelpers
+    def self.change_ownership(path, desired_owner, desired_group = nil)
+      path = Pathname.new(path)
+      return if Chef.node.platform_family?('windows')
+
+      require 'fileutils'
+      FileUtils.chown(desired_owner, desired_group, path.to_s)
+    end
+
+    def self.deep_change_ownership(path, owner, group = nil)
+      return if Chef.node.platform_family?('windows')
+
+      change_ownership(path, owner, group)
+      Pathname.glob(Pathname.new(path).join('**/*')).each { |sub_path| change_ownership(sub_path, owner, group) }
+    end
+  end unless defined? FileHelpers
```

Comment:
I go back and forth on this unless pattern when it comes to defining modules that are the base versions of the Module... Should we do this? Or would we accidentally break if someone wanted to monkeypatch in a method and got sourced before us.